### ISSUE TITLE
Invalid property type mapping for MyFaces org.apache.myfaces.CACHE_EL_EXPRESSIONS

### ISSUE THREAD

**Issue by jeansossmeier** [joinfaces/joinfaces#717]
Greetings,

The `org.joinfaces.autoconfigure.myfaces.MyfacesProperties.cacheElExpressions` is mapped as an `Boolean` while MyFaces expect an `org.apache.myfaces.view.facelets.ELExpressionCacheMode`

Expected EL Expression Cache Modes:
```
public enum ELExpressionCacheMode {
    noCache,
    strict,
    allowCset,
    always,
    alwaysRecompile;
}
```
Setting property `joinfaces.myfaces.cache-el-expressions` to the proper values becomes impossible and setting as a `Boolean` results in the following: 

![image](https://user-images.githubusercontent.com/5825385/62951347-36118880-bdc0-11e9-98d3-1b054aa10594.png)

**Comment by larsgrefer** [comment]
Thanks for the detailed bug report. This will be fixed in `4.0.8` and `4.1.0-m5`.

In the meantime you should be able to use this as workaround:

```properties
server.servlet.context-parameters.org.apache.myfaces.CACHE_EL_EXPRESSIONS=noCache
```

**Comment by jeansossmeier** [comment]
Thank you very much for your hard work in this great project.
